cartographer/io/serialization_format_migration_test.cc

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2018 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "cartographer/io/serialization_format_migration.h"
+
+#include <functional>
+#include <memory>
+
+#include "cartographer/io/internal/in_memory_proto_stream.h"
+#include "cartographer/mapping/proto/internal/legacy_serialized_data.pb.h"
+#include "cartographer/mapping/proto/pose_graph.pb.h"
+#include "cartographer/mapping/proto/serialization.pb.h"
+#include "cartographer/mapping/proto/trajectory_builder_options.pb.h"
+#include "gmock/gmock.h"
+#include "google/protobuf/text_format.h"
+#include "gtest/gtest.h"
+
+namespace cartographer {
+namespace io {
+namespace {
+
+using ::google::protobuf::TextFormat;
+using ::testing::Eq;
+using ::testing::SizeIs;
+
+class MigrationTest : public ::testing::Test {
+ protected:
+  void SetUp() {
+    writer_.reset(new ForwardingProtoStreamWriter(
+        [this](const google::protobuf::Message* proto) -> bool {
+          std::string msg_string;
+          TextFormat::PrintToString(*proto, &msg_string);
+          this->output_messages_.push_back(msg_string);
+          return true;
+        }));
+
+    mapping::proto::PoseGraph pose_graph;
+    mapping::proto::AllTrajectoryBuilderOptions all_options;
+    mapping::proto::LegacySerializedData submap;
+    submap.mutable_submap()->mutable_submap_id()->set_trajectory_id(0);
```

Comment:
Not sure if this makes a difference here, but you are setting default values here. (I believe this line has no effect.)